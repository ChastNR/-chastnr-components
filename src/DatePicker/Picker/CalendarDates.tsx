import { useCallback } from 'react';

import { useDatePickerContext } from './context';
import renderDay from './Day';

interface CalendarDatesProps {
  onSelect: (date: Date) => void;
}

const CalendarDates: React.FC<CalendarDatesProps> = ({ onSelect }) => {
  const { calendar, current, month, year, setCalendarDate } = useDatePickerContext();

  const onDayClick = useCallback(
    (value: Date) => {
      setCalendarDate(value);
      onSelect(value);
    },
    [onSelect, setCalendarDate]
  );

  return (
    <div className="cldr__tbl">{calendar.map(renderDay({ current, month, year, onDayClick }))}</div>
  );
};

export default CalendarDates;
