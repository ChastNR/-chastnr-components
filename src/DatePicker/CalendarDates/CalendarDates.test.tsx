import { fireEvent, render } from '@testing-library/react';
import { CalendarDates } from './CalendarDates';

const mockOnSelect = jest.fn();

jest.mock('../Context', () => {
  const currentDate = new Date();

  return {
    useDatePickerContext: jest.fn().mockReturnValue({
      calendar: [[1]],
      current: currentDate,
      month: currentDate.getMonth(),
      year: currentDate.getFullYear(),
      setCalendarDate: jest.fn(),
    }),
  };
});

describe('CalendarDates', () => {
  it('should render properly', () => {
    const calendarDates = render(<CalendarDates onSelect={mockOnSelect} />);

    expect(calendarDates.container.firstChild).toMatchSnapshot();
  });

  it('should call functions when day was clicked', () => {
    const calendarDates = render(<CalendarDates onSelect={mockOnSelect} />);

    const dayElement = calendarDates.container.querySelector('.cldr__day') as Element;

    fireEvent.click(dayElement, { bubbles: true });

    expect(mockOnSelect).toBeCalledTimes(1);
  });
});
