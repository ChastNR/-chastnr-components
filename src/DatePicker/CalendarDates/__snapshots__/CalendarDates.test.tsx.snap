// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDates should render properly 1`] = `
<div
  class="cldr__tbl"
>
  <div
    class="cldr__day"
  >
    1
  </div>
</div>
`;
